codec-redis/src/main/java/io/netty/handler/codec/redis/RedisMessageType.java

```diff
@@ -54,9 +55,32 @@ public boolean isInline() {
     }
 
     /**
-     * Return {@link RedisMessageType} for this type prefix {@code byte}.
+     * Determine {@link RedisMessageType} based on the type prefix {@code byte} read from given the buffer.
      */
-    public static RedisMessageType valueOf(byte value) {
+    public static RedisMessageType readFrom(ByteBuf in, boolean decodeInlineCommands) {
+        final int initialIndex = in.readerIndex();
+        final RedisMessageType type = valueOf(in.readByte());
```

Comment:
just use `in.getByte(in.readerIndex())` this way you will not need to reset the index

Reply:
How does that help? If I use the suggested method I instead have to manually advance the reader index in all other cases to "consume" the type indicator.

Reply:
never mind.. brain fart.